### PR TITLE
OAK-8494 - Support AWS Key Managed Service (SSE-KMS)

### PR DESCRIPTION
**Pull request by ankitaagar**
OAK-8494 - Support AWS Key Managed Service (SSE-KMS)

### CODE REVIEW

File: oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3DSWithSSEKMSwithKey.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ */
+public class TestS3DSWithSSEKMSwithKey extends TestS3Ds {
+
+        protected static final Logger LOG = LoggerFactory.getLogger(TestS3DSWithSSES3.class);
+
+        protected static long ONE_KB = 1024;
+        protected static long ONE_MB = ONE_KB * ONE_KB;
+        protected static long TEN_MB = ONE_MB * 10;
+        protected static long ONE_HUNDRED_MB = ONE_MB * 100;
+        protected static long ONE_GB = ONE_HUNDRED_MB * 10;
```
Comment (neerava) [inline oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3DSWithSSEKMSwithKey.java:40]:
```suggestion
        protected static long ONE_GB = ONE_MB * ONE_MB;
```

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 2]
```diff
@@ -939,10 +935,26 @@ private URI createPresignedURI(DataIdentifier identifier,
             final Date expiration = new Date();
             expiration.setTime(expiration.getTime() + expirySeconds * 1000);
 
-            GeneratePresignedUrlRequest request = new GeneratePresignedUrlRequest(bucket, key)
-                    .withMethod(method)
-                    .withExpiration(expiration);
-
+            GeneratePresignedUrlRequest request = null;
+            if (properties.getProperty(S3Constants.S3_ENCRYPTION)
```
Comment (nit0906) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java:939]: better to add a null check here

File: oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3DSWithSSEKMS.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        @Test
+        public void testInitiateDirectUploadUnlimitedURIs() throws DataRecordUploadException {
+            ConfigurableDataRecordAccessProvider ds = null;
+            try {
+                ds = (ConfigurableDataRecordAccessProvider) createDataStore();
+            } catch (RepositoryException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
```
Comment (nit0906) [inline oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3DSWithSSEKMS.java:91]: Let's not catch the exception here . Just throw it - that would make the test fail at this step only.

**Comment by ankitaagar** [comment]
Closing

**Pull request closed without merging**